var/spack/repos/builtin/packages/py-absl-py/package.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+
+
+class PyAbslPy(PythonPackage):
+    """
+    This repository is a collection of Python library code for building
+    Python applications.
+
+    The code is collected from Google's own Python code base, and has been
+    extensively tested and used in production.
+    """
+
+    homepage = "https://pypi.org/project/absl-py/"
+    url      = "https://pypi.io/packages/source/a/absl-py/absl-py-0.7.0.tar.gz"
+
+    version('0.7.0', '8718189e4bd6013bf79910b9d1cb0a76aecad8ce664f78e1144980fabdd2cd23')
+    version('0.1.6', '02c577d618a8bc0a2a5d1a51f160d3649745d7a2516d87025322f46ac1391a22', url="https://cosmo-pypi.phys.ethz.ch/simple/absl-py/0.1.6/absl-py-0.1.6.tar.gz")
```

Comment:
Why the different URL for the older version?  Can we get both from the same URL?

Reply:
Pypi is the canonical source for python packages, I prefer using their sources as the risk of one of these disappearing should be less than anyone’s “private” hosting. I can change it back if your prefer. 

I’m periodically going through all package.py’s that we are locally modifying and push things back upstream, while doing so I’m typically updating hashes to sha256 and adding the latest stable version (both to us and spack). For python packages that means going to the pypi page

Reply:
I haven’t checked, it was added as such in our fork. I’m only checking an merging stuff back to upstream spack. I’ll have a look next week